### ISSUE TITLE
Doesn't seem to be working with a custom devise set up

### ISSUE THREAD

**Issue by heaven** [apokalipto/devise_saml_authenticatable#8]
Hi, I have the next set up:
```ruby
constraints ->(request) { request.ssl? } do
  constraints :subdomain => /^(?:dev-)?(?:awards|profiles)$/ do
    devise_for :members, :path => '/',
      :only => [:sessions, :passwords, :confirmations, :omniauth_callbacks, :invitations],
      :controllers => { :omniauth_callbacks => 'omniauth_callbacks', :invitations => 'invitations' },
      :path_names => { :sign_in => 'login', :sign_out => 'logout', :password => 'secret' }
  end
end
```
And the steps from the read me don't work, no routes for saml/login.

**Comment by heaven** [comment]
If I define it as:
```ruby
devise_for :members, :path => '/'
```
The following error happens:
```
➜  crm git:(master) ✗ rails server
=> Booting Thin
=> Rails 4.1.7 application starting in development on http://0.0.0.0:3000
=> Run `rails server -h` for more startup options
=> Notice: server is listening on all interfaces (0.0.0.0). Consider using 127.0.0.1 (--binding option)
=> Ctrl-C to shutdown server
Exiting
/Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/route_set.rb:430:in `add_route': Invalid route name, already in use: 'new_member_session'  (ArgumentError)
You may have defined two routes with the same name using the `:as` option, or you may be overriding a route already defined by a resource with the same naming. For the latter, you can restrict the routes created with `resources` as explained here: 
http://guides.rubyonrails.org/routing.html#restricting-the-routes-created
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1495:in `add_route'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1472:in `decomposed_match'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1470:in `block in decomposed_match'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1368:in `block (2 levels) in member'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:763:in `scope'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1368:in `block in member'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1600:in `with_scope_level'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1364:in `member'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1470:in `decomposed_match'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1453:in `block in match'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1443:in `each'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1443:in `match'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:636:in `map_method'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:597:in `get'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/devise_saml_authenticatable-0.0.2/lib/devise_saml_authenticatable/routes.rb:5:in `block in devise_saml_authenticatable'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1147:in `block in resource'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1611:in `block (2 levels) in resource_scope'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:763:in `scope'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1611:in `block in resource_scope'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1600:in `with_scope_level'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1610:in `resource_scope'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:1146:in `resource'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/devise_saml_authenticatable-0.0.2/lib/devise_saml_authenticatable/routes.rb:4:in `devise_saml_authenticatable'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/devise-3.4.1/lib/devise/rails/routes.rb:249:in `block (4 levels) in devise_for'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/devise-3.4.1/lib/devise/rails/routes.rb:249:in `each'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/devise-3.4.1/lib/devise/rails/routes.rb:249:in `block (3 levels) in devise_for'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/devise-3.4.1/lib/devise/rails/routes.rb:452:in `with_devise_exclusive_scope'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/devise-3.4.1/lib/devise/rails/routes.rb:248:in `block (2 levels) in devise_for'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/devise-3.4.1/lib/devise/rails/routes.rb:351:in `block in devise_scope'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:887:in `block in constraints'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:763:in `scope'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:887:in `constraints'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/devise-3.4.1/lib/devise/rails/routes.rb:350:in `devise_scope'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/devise-3.4.1/lib/devise/rails/routes.rb:247:in `block in devise_for'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/devise-3.4.1/lib/devise/rails/routes.rb:223:in `each'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/devise-3.4.1/lib/devise/rails/routes.rb:223:in `devise_for'
	from /Users/heaven/workspace/crm/config/routes.rb:11:in `block (3 levels) in <top (required)>'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:887:in `block in constraints'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:763:in `scope'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:887:in `constraints'
	from /Users/heaven/workspace/crm/config/routes.rb:10:in `block (2 levels) in <top (required)>'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:887:in `block in constraints'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:763:in `scope'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/mapper.rb:887:in `constraints'
	from /Users/heaven/workspace/crm/config/routes.rb:8:in `block in <top (required)>'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/route_set.rb:337:in `instance_exec'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/route_set.rb:337:in `eval_block'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/actionpack-4.1.7/lib/action_dispatch/routing/route_set.rb:315:in `draw'
	from /Users/heaven/workspace/crm/config/routes.rb:1:in `<top (required)>'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/activesupport-4.1.7/lib/active_support/dependencies.rb:241:in `load'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/activesupport-4.1.7/lib/active_support/dependencies.rb:241:in `block in load'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/activesupport-4.1.7/lib/active_support/dependencies.rb:232:in `load_dependency'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/activesupport-4.1.7/lib/active_support/dependencies.rb:241:in `load'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/application/routes_reloader.rb:40:in `block in load_paths'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/application/routes_reloader.rb:40:in `each'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/application/routes_reloader.rb:40:in `load_paths'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/application/routes_reloader.rb:16:in `reload!'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/application/routes_reloader.rb:26:in `block in updater'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/activesupport-4.1.7/lib/active_support/file_update_checker.rb:75:in `call'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/activesupport-4.1.7/lib/active_support/file_update_checker.rb:75:in `execute'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/application/routes_reloader.rb:27:in `updater'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/application/routes_reloader.rb:7:in `execute_if_updated'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/application/finisher.rb:71:in `block in <module:Finisher>'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/initializable.rb:30:in `instance_exec'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/initializable.rb:30:in `run'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/initializable.rb:55:in `block in run_initializers'
	from /Users/heaven/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:226:in `block in tsort_each'
	from /Users/heaven/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:348:in `block (2 levels) in each_strongly_connected_component'
	from /Users/heaven/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:427:in `each_strongly_connected_component_from'
	from /Users/heaven/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:347:in `block in each_strongly_connected_component'
	from /Users/heaven/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:345:in `each'
	from /Users/heaven/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:345:in `call'
	from /Users/heaven/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:345:in `each_strongly_connected_component'
	from /Users/heaven/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:224:in `tsort_each'
	from /Users/heaven/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:205:in `tsort_each'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/initializable.rb:54:in `run_initializers'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/application.rb:300:in `initialize!'
	from /Users/heaven/workspace/crm/config/environment.rb:5:in `<top (required)>'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/activesupport-4.1.7/lib/active_support/dependencies.rb:247:in `require'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/activesupport-4.1.7/lib/active_support/dependencies.rb:247:in `block in require'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/activesupport-4.1.7/lib/active_support/dependencies.rb:232:in `load_dependency'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/activesupport-4.1.7/lib/active_support/dependencies.rb:247:in `require'
	from /Users/heaven/workspace/crm/config.ru:3:in `block in <main>'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/rack-1.5.2/lib/rack/builder.rb:55:in `instance_eval'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/rack-1.5.2/lib/rack/builder.rb:55:in `initialize'
	from /Users/heaven/workspace/crm/config.ru:in `new'
	from /Users/heaven/workspace/crm/config.ru:in `<main>'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/rack-1.5.2/lib/rack/builder.rb:49:in `eval'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/rack-1.5.2/lib/rack/builder.rb:49:in `new_from_string'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/rack-1.5.2/lib/rack/builder.rb:40:in `parse_file'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/rack-1.5.2/lib/rack/server.rb:277:in `build_app_and_options_from_config'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/rack-1.5.2/lib/rack/server.rb:199:in `app'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/commands/server.rb:50:in `app'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/rack-1.5.2/lib/rack/server.rb:314:in `wrapped_app'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/commands/server.rb:130:in `log_to_stdout'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/commands/server.rb:67:in `start'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/commands/commands_tasks.rb:81:in `block in server'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/commands/commands_tasks.rb:76:in `tap'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/commands/commands_tasks.rb:76:in `server'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/commands/commands_tasks.rb:40:in `run_command!'
	from /Users/heaven/.rvm/gems/ruby-2.1.2@carmen/gems/railties-4.1.7/lib/rails/commands.rb:17:in `<top (required)>'
	from bin/rails:10:in `require'
	from bin/rails:10:in `<main>'
➜
```

**Comment by stefanahman** [comment]
+1

**Comment by apokalipto** [comment]
Closing this issue because this error is related to devise and rails 4. See https://github.com/plataformatec/devise/issues/2507.

**Comment by archonic** [comment]
Am I right to understand that running this gem and having custom devise routes with `devise_for` simply won't work? I'm not sure what the work around should be.

**Comment by adamstegman** [comment]
@archonic is there a specific problem you're running into? I'm not aware of any issues myself.

**Comment by archonic** [comment]
In my app I have the following in routes.rb:
```
devise_for :users, path: "",
      path_names: { sign_in: "login", sign_out: "logout", registration: "profile" },
      controllers: { registrations: "users/registrations", invitations: "users/invitations", sessions: "users/sessions", passwords: "users/passwords", unlocks: "users/unlocks" }
```
This was conflicting with the dynamic route (new_user_session) in this gem. I believe I've solved it with the suggestion in #43 - https://github.com/archonic/devise_saml_authenticatable/commit/88a2f1627cdd0dc65878f3bc6805b1d48f3b4322
